src/typeahead/typeahead.ts

```diff
@@ -194,11 +204,39 @@ export class NgbTypeahead implements OnInit,
     }
   }
 
+  private _closePopup() {
+    this._popupService.close();
+    this._windowRef = null;
+  }
+
   private _selectResult(result: any) {
     this.writeValue(result);
     this._onChangeNoEmit(result);
     this.selectItem.emit(result);
-    this.closePopup();
+    this._closePopup();
+  }
+
+  private _showHint() {
+    if (this.showHint) {
+      const userInputLowerCase = this._userInput.toLowerCase();
+      const formattedVal = this._formatItemForInput(this._windowRef.instance.getActive());
+
+      if (userInputLowerCase === formattedVal.substr(0, this._userInput.length).toLowerCase()) {
+        this._writeInputValue(this._userInput + formattedVal.substr(this._userInput.length));
+        this._renderer.invokeElementMethod(
+            this._elementRef.nativeElement, 'setSelectionRange', [this._userInput.length, formattedVal.length]);
+      } else {
+        this.writeValue(this._windowRef.instance.getActive());
+      }
+    }
+  }
+
+  private _formatItemForInput(item: any): string {
+    return item && this.inputFormatter ? this.inputFormatter(item) : toString(item);
```

Comment:
I know we're expecting `this.inputFormatter` to be a function here, but this could be problematic if a user is using ES6 - are we fine with there being an error in that situation?